test/integration/auction.test.js

```diff
@@ -62,7 +85,152 @@ describe('Auctions', function () {
 
           expect(res).to.have.status(200)
           expect(res.body).to.be.a('object')
-          expect(res.body.seller).to.be.eql(userData[0].username)
+          expect(res.body.seller).to.be.equal(userData[0].username)
+
+          return done()
+        })
+    })
+
+    it('Should return 404, for Auction not found', function (done) {
+      chai
+        .request(app)
+        .get('/auctions/invalidId')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.equal('Auction doesn\'t exist')
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions', function () {
+    it('Should create a new auction', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send(auctionData)
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Auction created successfully!')
+
+          return done()
+        })
+    })
+
+    it('Should have enough Item Type in wallet', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ ...auctionData, quantity: 5 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal(`You do not have enough of ${itemType}s!`)
+
+          return done()
+        })
+    })
+
+    it('Should return 404, for Auction not Found', function (done) {
+      chai
+        .request(app)
+        .post('/auction')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.deep.equal({
+            statusCode: 404,
+            error: 'Not Found',
+            message: 'Not Found'
+          })
+
+          return done()
+        })
+    })
+  })
```

Comment:
Again, what is this doing? 🤔

Reply:
Same

Reply:
For this the `statusCode : 401` and it is changed in the local.
Route: `/auctions`, for Unathorized user